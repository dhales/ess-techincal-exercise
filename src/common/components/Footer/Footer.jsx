import React, { Fragment, useContext } from 'react';

import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';

import { UserContext } from '../../state/context';

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  title: {
    flexGrow: 1,
  },
}));

export default function Footer() {
  const classes = useStyles();
  const {user, updateUser} = useContext(UserContext);

  
  const logOut = () => {
    user.auth = false;
    user.userName = "";
    user.selectedAssociate = {
      name:null,
      email:null,
    }
    updateUser({user});
  }

  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar>
          {user && (
            <Fragment>
              <Typography variant="h6" className={classes.title}>
                {user.selectedAssociate.name && (`Associate selected: ${user.selectedAssociate.name}, ${user.selectedAssociate.email}`)}
              </Typography>
              {user.auth && (
                <Button onClickCapture={logOut} variant="contained">Logout</Button>
              )}
            </Fragment>
          )}
        </Toolbar>
      </AppBar>
    </div>
  );
}
