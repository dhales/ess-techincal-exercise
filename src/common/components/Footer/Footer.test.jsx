import React from 'react';

import {render} from '@testing-library/react';

import { UserContext } from '../../state/context';

import Footer from './Footer';

const user = {
    userName:null, 
    selectedAssociate: {
        name:null,
        email:null,
    },
    auth:false,
}

const updateUser = () => null;

describe("Footer", () => {
    it('Should render Footer', () => {
        const { container } = render(
            <UserContext.Provider value={user, updateUser}>
                <Footer />
            </UserContext.Provider>
            
        );
        expect(container).toMatchSnapshot();
    });
});