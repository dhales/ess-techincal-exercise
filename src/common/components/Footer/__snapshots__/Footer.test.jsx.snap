// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Should render Footer 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      />
    </header>
  </div>
</div>
`;
